packages/flutter/lib/src/widgets/router.dart

```diff
@@ -297,11 +295,11 @@ class Router<T> extends StatefulWidget {
 
   /// Retrieves the immediate [Router] ancestor from the given context.
   ///
-  /// Use this method when you need to access the delegates in the [Router].
-  /// For example, you need to access the [backButtonDispatcher] of the parent
-  /// router to create a [ChildBackButtonDispatcher] for a nested router.
-  /// Another use case may be updating the value in [routeInformationProvider]
-  /// to navigate to a new route.
+  /// This method provides access to the delegates in the [Router]. For example,
+  /// this can be used to access the [backButtonDispatcher] of the parent router
+  /// when creating a [ChildBackButtonDispatcher] for a nested [Router].
+  /// Similarly, it might be used to cause the [routeInformationProvider] to
```

Comment:
Oops, this is a false statement, the routeInformationProvider does not guarantee to have value setter. I originally added this statement when I added value setter in PlatformRouteInformationProvider, and it was removed as a part of addressing pr reviews, I forgot to remove the comment here.

Reply:
Sorry can you clarify exactly what I should change? I didn't quite follow your description.

Reply:
I mean we should remove line 301 and line 302.

basically there is not API in routeInformationProvider to change the route by default.

Reply:
got it. done.